cordapp-cpk2/src/main/java/net/corda/plugins/cpk2/CordappPlugin.java

```diff
@@ -526,8 +539,22 @@ private void configureCordappAttributes(@NotNull String symbolicName, @NotNull A
             attributes.put("Cordapp-Contract-Vendor", vendor);
             attributes.put("Cordapp-Contract-Licence", licence);
         }
+
         final CordappData workflow = cordapp.getWorkflow();
         if (!workflow.isEmpty()) {
+            if (!setName) {
```

Comment:
This code (what was here originally) looks a little peculiar. It's saying if there was a workflow use that as the bundle description, otherwise use the contract. I wasn't aware you could define both in the same bundle. Contract bundles are loaded into verification sandboxes and workflow bundles are loaded into flow sandboxes. As far as I know we shouldn't see both kinds of classes in the same bundle at the same time.

I kind of assumed when looking at the README if you defined both in the same module you got 2 bundles.

I also see this function is called `configureCordappAttributes` which suggests it's setting these at app scope and not CPK scope.

Reply:
Ok, having unpacked a CPB it looks as I'd expect, i.e. CPB contains only
```
Corda-CPB-Format: 2.0
Corda-CPB-Name: packaging-verification-app-v1
Corda-CPB-Version: 5.0.0.0-SNAPSHOT
```

And each CPK (contract and workflow) jar contains (amongst other things):
```
Corda-CPK-Cordapp-Name: net.corda.packaging-verification-app-v1
```
and
```
Corda-CPK-Cordapp-Name: net.corda.packaging-verification-contract-v1
```

So why the plugin suggests these are cordapp wide attributes is beyond me.

Reply:
For now I'd just remove the `if (!setName)` check, and have it overwrite. At least then it's consistent with the other attributes. Then you could roll the `attributes.put(CPK_CORDAPP_NAME, symbolicName);` part into the other blocks as @kyriathar suggests below.